docs/v1/api.md

```diff
@@ -1,63 +1,96 @@
+   (`GET /v2/catalog`)
+2. Translates the response to a list of `ServiceClass`
+3. Creates new `ServiceClass` instances in the API server
 
-## `Binding`
+*TODO: What happens when a `Broker` resource is deleted?*
 
-This resource is created by a service consumer to indicate that an application
-should be bound to an instance. When the service catalog's controller event
-loop sees a `Binding` created, it calls the bind endpoint on the backing CF
-service broker. When a successful response is returned, the event loop does
-the following:
+## `ServiceClass` Resource
 
-1. Writes `bound` into a `status` field of the `Binding`
-1. Writes the contents of `credentials` map into a secret (naming of the secret
-   to be discussed later)
-1. Updates the `Binding`'s status section with the fully qualified path to the
-   aforementioned secret
+A `ServiceClass` represents an offering in the service catalog.  A
+`ServiceClass` is not usable directly; an `Instance` of a `ServiceClass` must be
+created to be consumed by an application.
 
-*TODO: what happens when a `Binding` resource is deleted?*
+This resource is created by the service catalog's controller event loop after
+it has received a `Broker` resource and successfully called the backing CF
+broker's catalog endpoint.
+
+*TODO: Make explicit the relationship between service classes and plans*
+
+## `Instance` Resource
+
+An `Instance` represents a provisioned instance of a `ServiceClass`, and is the
+entity an application binds to.
+
+A service consumer creates an `Instance` to indicate their intent to provision
+an instance of a service class.  The `Instance` has a reference to the
+`ServiceClass` to provision an instance of.
+
+The service catalog controller has a watch on the `Instance` resource and
+receives an `ADD` watch event. The controller then attempts to provision a new
+instance:
+
+1.  The controller calls the provision endpoint on the broker 
+2.  The broker returns a response indicating that the broker provisioned the new
+    instance, the instance was already provisioned, or the provisioning
+    operation is in progress
+3.  The controller updates the status of the `Instance` to indicate when the
+    `Instance` is in a provisioning or provisioned condition
+
+*TODO: codify how asynchronous responses are handled in the controller*
+
+## `Binding` Resource
+
+A `Binding` represents the binding between an application and an `Instance` of a
+`ServiceClass`.  
+
+Service consumers create `Binding` resources to indicate that an application
+should be bound to an `Instance`.  The `Binding` contains information about how
+the application wants to use the binding information such as:
+
+1.  The name of a Kubernetes core `Service` resource to provide a stable
+    endpoint for the application to use the `Instance` via
+2.  The name of a Kubernetes `Secret` resource to hold credentials necessary to
+    use the service
+3.  The name and spec of a Kubernetes `ServiceInjectionPolicy` to create
```

Comment:
There's no section for `ServiceInjectionPolicy` in this doc. Although we've talked about it, let's leave this point out or create a section for it here.

Since this resource and its admission controller must both exist, and neither are yet available, I'd prefer that we do the former.